### PR TITLE
Removing exception on table with no columns during get table call

### PR DESCRIPTION
**Pull request by fal-bharadwaj**
*Issue #, if available:*
https://github.com/awslabs/aws-athena-query-federation/issues/2699

*Description of changes:*
This PR fixes the issue that currently exists where query editor window does not load any tables if another table in the db is empty and instead shows an exception (as described in the bug https://github.com/awslabs/aws-athena-query-federation/issues/2699). Verified the fix in dev account.

Implemented the following changes -
- `athena-jdbc->JdbcMetadataHandler`: Removed the exception in getSchema call for when a table has no columns, instead added a log warning for it. Moved the partition schema updates to happen only for tables where columns exist since we're not exiting out of the call due to the exception anymore.
- `athena-jdbc->JdbcMetadataHandlerTest`: Updated the test to account for this change.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: athena-jdbc/src/main/java/com/amazonaws/athena/connectors/jdbc/manager/JdbcMetadataHandler.java [hunk 1]
```diff
@@ -414,12 +414,13 @@ protected Schema getSchema(Connection jdbcConnection, TableName tableName, Schem
             }
 
             if (!found) {
-                throw new AthenaConnectorException(String.format("Could not find table %s in %s", tableName.getTableName(), tableName.getSchemaName()),
-                        ErrorDetails.builder().errorCode(FederationSourceErrorCode.ENTITY_NOT_FOUND_EXCEPTION.toString()).build());
+                // log a warning if table columns are not found
+                LOGGER.warn("getSchema: Could not find table " + tableName.getTableName() + " in " + tableName.getSchemaName());
```
[the remark this answers is not in the document]
Comment (AbdulR3hman) [reply athena-jdbc/src/main/java/com/amazonaws/athena/connectors/jdbc/manager/JdbcMetadataHandler.java:418]: better way is to pass the variable directly to the warn function; it knows how to treat it; see below:

```
LOGGER.warn("getSchema: Could not find table {} in {}", 
            tableName.getTableName(), 
            tableName.getSchemaName());
```

**Review by AbdulR3hman — CHANGES_REQUESTED** [review]

**Review by aimethed — APPROVED** [review]

**Review by AbdulR3hman — APPROVED** [review]
approved; but if you can also add a test case to make sure that the returned schema is empty that would be awesome.

**Review by AbdulR3hman — APPROVED** [review]
